### DISCUSSION TITLE
Parse string into Date type

### DISCUSSION THREAD

**Discussion by Tyrubias** [MichalLytek/type-graphql#807] [Q&A] [answered]
I'm wrapping a REST API into a GraphQL endpoint. The REST API returns a `string` where I want a `Date`. If I try to declare the field as a `Date`, I get the error `Unable to serialize value '2020-10-13T08:17:46.832Z' as it's not an instance of 'Date`.

My schema:

```ts
@ObjectType()
class Widget {
    @Field(() => Date)
    updatedAt: string
}
```

How should I get GraphQL to parse the `Date` string and not serialize it?

**Comment by MichalLytek** [comment, marked as answer]
`Date` is `Date`, the JS `new Date()`.
If you want to treat string as date, just do `@Field(() => String)` or find another GraphQL scalar that accepts strings and numbers as a valid date value.
JSON is not capable of transporting `Date` type, it has to be serialized.
